### PR TITLE
4857: Adding account payments endpoint to return all failed payments for NSF

### PR DESCRIPTION
**Pull request by sumesh-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4857

*Description of changes:*
- 4857: Adding account payments endpoint to return all failed payments for NSF

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updated the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/resources/invoice.py [hunk 1]
```diff
@@ -11,38 +11,92 @@
...
 
-API = Namespace('invoices', description='Payment System - Payment Requests')
 
+API = Namespace('invoice', description='Payment System - Invoices')
 
-@cors_preflight(['GET'])
-@API.route('', methods=['GET', 'OPTIONS'], doc={'deprecated': True})
-class PaymentRequestInvoice(Resource):
-    """Temporary endpoint to unblock teams who are using this endpoint."""
+
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/resources/invoice.py:33]: This file has been renamed from payment to invoice.py.. diff doesn't reflect this.

File: pay-api/src/pay_api/resources/invoices.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from pay_api.exceptions import BusinessException
+from pay_api.services import InvoiceService
+from pay_api.utils.auth import jwt as _jwt
+from pay_api.utils.trace import tracing as _tracing
+from pay_api.utils.util import cors_preflight
+
+API = Namespace('invoices', description='Payment System - Payment Requests')
+
+
+@cors_preflight(['GET'])
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/resources/invoices.py:29]: This has been renamed to invoices.py from invoice.py

File: pay-api/src/pay_api/resources/payment.py [hunk 3]
```diff
@@ -11,92 +11,40 @@
...
-        current_app.logger.info('<Payment.post')
-        request_json = request.get_json()
-        current_app.logger.debug(request_json)
-        # Validate the input request
-        valid_format, errors = schema_utils.validate(request_json, 'payment_request')
-
-        if not valid_format:
-            return error_to_response(Error.INVALID_REQUEST, invalid_params=schema_utils.serialize(errors))
-
+    def get(account_id: str):
```
Comment (sumesh-aot) [inline pay-api/src/pay_api/resources/payment.py:39]: Using payment.py for /payments endpoint, not for payment requests

File: pay-api/migrations/versions/b05d371ad030_payment_failed_status.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# revision identifiers, used by Alembic.
+revision = 'b05d371ad030'
+down_revision = 'b7443d501d98'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('payment', sa.Column('cons_inv_number', sa.String(length=50), nullable=True))
```
Comment (saravanpa-aot) [inline pay-api/migrations/versions/b05d371ad030_payment_failed_status.py:22]: is this cons_inv_number in the model?somehow i cant see it..

Comment (sumesh-aot) [reply pay-api/migrations/versions/b05d371ad030_payment_failed_status.py:22]: That was from my changes for consolidated invoices, shouldn't impact any existing features.

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**